homeassistant/components/ambee/sensor.py

```diff
@@ -73,15 +55,13 @@ def __init__(
         self._service_key = service_key
 
         self.entity_id = f"{SENSOR_DOMAIN}.{service_key}_{sensor_key}"
-        self._attr_device_class = sensor.get(ATTR_DEVICE_CLASS)
-        self._attr_entity_registry_enabled_default = sensor.get(
-            ATTR_ENABLED_BY_DEFAULT, True
-        )
-        self._attr_icon = sensor.get(ATTR_ICON)
-        self._attr_name = sensor.get(ATTR_NAME)
-        self._attr_state_class = sensor.get(ATTR_STATE_CLASS)
+        self._attr_device_class = sensor.device_class
```

Comment:
```suggestion
        self._attr_device_class = metadata.device_class
```
I would also suggest to rename the parameter from `sensor` to `metadata`, just for clarity.

Reply:
Nope. See above.